explorer/ast/declaration.h

```diff
@@ -126,6 +126,59 @@ class Declaration : public AstNode {
   bool is_type_checked_ = false;
 };
 
+// A name being declared in a named declaration.
+class DeclaredName {
+ public:
+  struct NameComponent {
+    SourceLocation source_loc;
+    std::string name;
+  };
+
+  // This is required by bison, and should not be used elsewhere.
+  explicit DeclaredName() {}
+
+  explicit DeclaredName(SourceLocation loc, std::string name)
```

Comment:
```suggestion
 DeclaredName(SourceLocation loc, std::string name)
```

Reply:
This was intentional -- I don't really want `{loc, name}` to be usable to implicitly create instances -- though I think it's pretty unimportant in the grand scheme of things. The C++ style guide says only "Constructors that cannot be called with a single argument may omit explicit." which this is in line with.

Reply:
I think the code rarely has `explicit` on non-signle-argument constructors, and more often as a mistake than deliberate. I don't think that DeclaredName and NamespaceDeclaration (which I'd just missed) should be different because it's a hard style divergence to infer.

To be clear here, I think this is about consistency inside the project, not explicit style. I would actually assume it'll get removed on some misc cleanup pass removing `explicit`, just because it's a divergence.

I'll note, if there's a strong reason for this, it really needs comments to explain the need so that it's not inadvertently removed.

Reply:
I think the right thing to do here is make sure there's a consensus on changing project style. I've brought this up on https://discord.com/channels/655572317891461132/821113559755784242/1070759074112741447